src/src/PluginStructs/P104_data_struct.cpp

```diff
@@ -0,0 +1,2458 @@
+  }
+  # endif // ifdef P104_USE_DATETIME_OPTIONS
+
+  addFormSubHeader(F("Zones"));
+
+  { // Zones
+    String zonesList[P104_MAX_ZONES];
+    int    zonesOptions[P104_MAX_ZONES];
+
+    for (uint8_t i = 0; i < P104_MAX_ZONES; i++) {
+      zonesList[i]    = i + 1;
+      zonesOptions[i] = i + 1; // No 0 needed or wanted
+    }
+    # if defined(P104_USE_TOOLTIPS) || defined(P104_ADD_SETTINGS_NOTES)
+    String zonetip;
+
+    if (zonetip.reserve(90)) {
+      zonetip  = F("Select between 1 and ");
+      zonetip += P104_MAX_ZONES;
+      zonetip += F(" zones, changing");
+      #  ifdef P104_USE_ZONE_ORDERING
+      zonetip += F(" Zones or Zone order");
+      #  endif // ifdef P104_USE_ZONE_ORDERING
+      zonetip += F(" will save and reload the page.");
+    }
+    # endif    // if defined(P104_USE_TOOLTIPS) || defined(P104_ADD_SETTINGS_NOTES)
+    addFormSelector(F("Zones"), F("plugin_104_zonecount"), P104_MAX_ZONES, zonesList, zonesOptions, NULL, P104_CONFIG_ZONE_COUNT, true
+                    # ifdef P104_USE_TOOLTIPS
+                    , zonetip
+                    # endif // ifdef P104_USE_TOOLTIPS
+                    );
+
+    # ifdef P104_USE_ZONE_ORDERING
+    const String orderTypes[] = {
+      F("Numeric order (1..n)"),
+      F("Display order (n..1)")
+    };
+    const int    orderOptions[] = { 0, 1 };
+    addFormSelector(F("Zone order"), F("plugin_104_zoneorder"), 2, orderTypes, orderOptions, NULL,
+                    bitRead(P104_CONFIG_FLAGS, P104_CONFIG_FLAG_ZONE_ORDER) ? 1 : 0, true
+                    #  ifdef P104_USE_TOOLTIPS
+                    , zonetip
+                    #  endif // ifdef P104_USE_TOOLTIPS
+                    );
+    # endif                  // ifdef P104_USE_ZONE_ORDERING
+    # ifdef P104_ADD_SETTINGS_NOTES
+    addFormNote(zonetip);
+    # endif // ifdef P104_ADD_SETTINGS_NOTES
+  }
+  expectedZones = P104_CONFIG_ZONE_COUNT;
+
+  if (expectedZones == 0) { expectedZones++; } // Minimum of 1 zone
+
+  { // Optionlists and zones table
+    const __FlashStringHelper *alignmentTypes[3] = {
+      F("Left"),
+      F("Center"),
+      F("Right")
+    };
+    const int alignmentOptions[3] = {
+      static_cast<int>(textPosition_t::PA_LEFT),
+      static_cast<int>(textPosition_t::PA_CENTER),
+      static_cast<int>(textPosition_t::PA_RIGHT)
+    };
+    int animationCount = 6;
+    # if ENA_SPRITE
+    animationCount += 1;
+    # endif // ENA_SPRITE
+    # if ENA_MISC
+    animationCount += 6;
+    # endif // ENA_MISC
+    # if ENA_WIPE
+    animationCount += 2;
+    # endif // ENA_WIPE
+    # if ENA_SCAN
+    animationCount += 4;
+    # endif // ENA_SCAN
+    # if ENA_OPNCLS
+    animationCount += 4;
+    # endif // ENA_OPNCLS
+    # if ENA_SCR_DIA
+    animationCount += 4;
+    # endif // ENA_SCR_DIA
+    # if ENA_GROW
+    animationCount += 2;
+    # endif // ENA_GROW
+    String animationTypes[] {
```

Comment:
This one can be of type flash string

Reply:
No, as each string gets the ordinal number appended in a loop, somewhat below. around line 2000.

Reply:
Ah OK.
Maybe put that in a separate scope then (`{}`) as it is quite a long array of `String`